node/src/components/linear_chain_sync/operations.rs

```diff
@@ -588,3 +612,179 @@ where
 
     Ok(trusted_header_output)
 }
+
+/// Returns `true` if `most_recent_block` belongs to an era that is still ongoing.
+fn is_current_era<I>(
+    most_recent_block: &BlockHeader,
+    maybe_switch_block: Option<&BlockHeader>,
+    chainspec: &Chainspec,
+    now: Timestamp,
+) -> Result<bool, LinearChainSyncError<I>>
+where
+    I: Eq + Debug,
+{
+    // Compute the start timestamp of the current era, and the number of blocks so far.
+    let (blocks_in_this_era, era_start) = match maybe_switch_block {
+        Some(switch_block) => (
+            most_recent_block.height() - switch_block.height(),
+            switch_block.timestamp(),
+        ),
+        None => (
+            most_recent_block.height() + 1,
+            chainspec
+                .protocol_config
+                .activation_point
+                .genesis_timestamp()
+                .ok_or_else(|| {
+                    error!(
+                        ?most_recent_block,
+                        "no recent switch block and no genesis timestamp"
+                    );
+                    LinearChainSyncError::MissingGenesisTimestamp
+                })?,
+        ),
+    };
+
+    // If the minimum era duration has not yet run out, the era is still current.
+    if now.saturating_diff(era_start) < chainspec.core_config.era_duration {
```

Comment:
In the `linear_chain_sync` we use this to compute the shortest era:
```rust
// Shortest era is the maximum of the two.
            let shortest_era: TimeDiff = std::cmp::max(
                chainspec.highway_config.min_round_length()
                    * chainspec.core_config.minimum_era_height,
                chainspec.core_config.era_duration,
            );
```

Reply:
Yes, but TBH I find it a bit hard to reason about that code. My implementation should be equivalent, but it separates the era duration criterion from the minimum height, and doesn't compute anything in advance.
The `shortest_era` computation above is just half of the logic in `linear_chain_sync.rs`, and the other half is in [`is_currently_active_era`](https://github.com/casper-network/casper-node/blob/b539bd0c76f76d53bae2162199659d1dd4f4a85b/node/src/components/linear_chain_sync.rs#L392).

(But maybe it's just me; if you prefer the other approach I'm happy to use that instead.)